brainscore_vision/model_helpers/activations/core.py

```diff
@@ -53,38 +97,46 @@ def from_stimulus_set(self, stimulus_set, layers, stimuli_identifier=None):
+                                    stimuli_identifier=stimuli_identifier,
+                                    require_variance=require_variance)
         else:
             self._logger.debug(f"self.identifier `{self.identifier}` or stimuli_identifier {stimuli_identifier} "
                                f"are not set, will not store")
             fnc = self._from_paths
-        # In case stimuli paths are duplicates (e.g. multiple trials), we first reduce them to only the paths that need
-        # to be run individually, compute activations for those, and then expand the activations to all paths again.
-        # This is done here, before storing, so that we only store the reduced activations.
-        reduced_paths = self._reduce_paths(stimuli_paths)
-        activations = fnc(layers=layers, stimuli_paths=reduced_paths)
-        activations = self._expand_paths(activations, original_paths=stimuli_paths)
+        if require_variance:
+            activations = fnc(layers=layers, stimuli_paths=stimuli_paths, require_variance=require_variance)
+        else:
+            # In case stimuli paths are duplicates (e.g. multiple trials), we first reduce them to only the paths that need
+            # to be run individually, compute activations for those, and then expand the activations to all paths again.
+            # This is done here, before storing, so that we only store the reduced activations.
+            reduced_paths = self._reduce_paths(stimuli_paths)
+            activations = fnc(layers=layers, stimuli_paths=reduced_paths, require_variance=require_variance)
+            activations = self._expand_paths(activations, original_paths=stimuli_paths)
         return activations
 
     @store_xarray(identifier_ignore=['stimuli_paths', 'layers'], combine_fields={'layers': 'layer'})
-    def _from_paths_stored(self, identifier, layers, stimuli_identifier, stimuli_paths):
+    def _from_paths_stored(self, identifier, layers, stimuli_identifier,
+                           stimuli_paths, number_of_trials: int = 1, require_variance: bool = False):
         return self._from_paths(layers=layers, stimuli_paths=stimuli_paths)
 
-    def _from_paths(self, layers, stimuli_paths):
+    def _from_paths(self, layers, stimuli_paths, require_variance: bool = False):
         if len(layers) == 0:
             raise ValueError("No layers passed to retrieve activations from")
         self._logger.info('Running stimuli')
-        layer_activations = self._get_activations_batched(stimuli_paths, layers=layers, batch_size=self._batch_size)
+        layer_activations = self._get_activations_batched(stimuli_paths, layers=layers, batch_size=self._batch_size,
+                                                          require_variance=require_variance)
         self._logger.info('Packaging into assembly')
-        return self._package(layer_activations, stimuli_paths)
+        return self._package(layer_activations, stimuli_paths, require_variance)
```

Comment:
whenever there's more than 1-2 arguments, I prefer to use keywords to avoid confusion and inadvertent mis-ordering.
```suggestion
        return self._package(layer_activations=layer_activations, stimuli_paths=stimuli_paths, require_variance=require_variance)
```